src/utils/scroll-and-highlight.tsx

```diff
@@ -0,0 +1,27 @@
+export const scrollToAndHighlight = (
+  path: string,
+  isIncreasedHeight: boolean = true
+) => {
+  setTimeout(() => {
+    const elementWithId = document.getElementById(
+      path.split('#').slice(1).join()
+    )
+    const highlightedElement = elementWithId?.querySelectorAll('div')[0]

```

Comment:
Direct DOM manipulation is not recommended in React. It makes the code less readable and maintainable. You should embrace declarative approach. If you need access to DOM elements you should use Refs. It would be useful to read this: https://react.dev/learn/manipulating-the-dom-with-refs